src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.Variables.cs

```diff
@@ -0,0 +1,373 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal partial class NullableWalker
+    {
+        /// <summary>
+        /// An immutable copy of Variables.
+        /// </summary>
+        [DebuggerDisplay("{GetDebuggerDisplay(), nq}")]
+        internal sealed class VariablesSnapshot
+        {
+            internal readonly int Id;
+            internal readonly VariablesSnapshot? Container;
```

Comment:
>Container [](start = 49, length = 9)

Is there some rule about how slots are distributed across the hierarchy? I think it would be good to document and assert them.

Reply:
Documented in `Variables` type.

---
In reply to: [561456595](https://github.com/dotnet/roslyn/pull/50417#discussion_r561456595) [](ancestors = 561456595)

Reply:
> Documented in Variables type.

Consider adding this information to the doc comment. 

---
In reply to: [562119139](https://github.com/dotnet/roslyn/pull/50417#discussion_r562119139) [](ancestors = 562119139,561456595)